From b94fb07f068103ad68901b7440d7d82cfe4b18ba Mon Sep 17 00:00:00 2001
From: Stephane Gasparini <stephane.gasparini@intel.com>
Date: Mon, 1 Feb 2016 17:09:25 +0100
Subject: [PATCH] kernel/fs: fix I/O wait not accounted for RW O_DSYNC

 When a process is doing Random Write with O_DSYNC flag
 the I/O wait are not accounted in the kernel (get_cpu_iowait_time_us).
 This is preventing the governor or the cpufreq driver to account for
 I/O wait and thus use the right pstate

Change-Id: I6b165d5d34634871eebac8a242dceb068010677f
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18385
Signed-off-by: Stephane Gasparini <stephane.gasparini@intel.com>
Reviewed-on: https://android.intel.com:443/464887
---
 fs/bio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/bio.c b/fs/bio.c
index b2b1451..86722b1 100644
--- a/fs/bio.c
+++ b/fs/bio.c
@@ -882,7 +882,7 @@ int submit_bio_wait(int rw, struct bio *bio)
 	bio->bi_private = &ret;
 	bio->bi_end_io = submit_bio_wait_endio;
 	submit_bio(rw, bio);
-	wait_for_completion(&ret.event);
+	wait_for_completion_io(&ret.event);
 
 	return ret.error;
 }
-- 
1.9.1

